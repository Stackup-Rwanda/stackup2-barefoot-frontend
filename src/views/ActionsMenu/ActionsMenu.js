import React from 'react';
import PropTypes, { object } from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import MoreVert from '@material-ui/icons/MoreVert';

const useStyles = makeStyles({
  tableIcons: {
    fontSize: 16,
  },
  item: {
    fontSize: 12,
    color: '#484848',
  },
});

const ActionsMenu = ({ actions }) => {
  const [anchorEl, setAnchorEl] = React.useState(null);

  const handleClick = (event) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  const classes = useStyles();

  return (
    <div>
      <Button aria-controls="simple-menu" aria-haspopup="true" onClick={handleClick}>
        <MoreVert className={classes.tableIcons} htmlColor="#00A799" />
      </Button>
      <Menu
        id="simple-menu"
        anchorEl={anchorEl}
        keepMounted
        open={Boolean(anchorEl)}
        onClose={handleClose}
      >
        {
          actions.map((action) => (
            <MenuItem key={action.id} className={classes.item} onClick={handleClose}>
              {action.name}
            </MenuItem>
          ))
        }
      </Menu>
    </div>
  );
};

ActionsMenu.propTypes = {
  actions: PropTypes.arrayOf(object).isRequired,
};

export default ActionsMenu;
