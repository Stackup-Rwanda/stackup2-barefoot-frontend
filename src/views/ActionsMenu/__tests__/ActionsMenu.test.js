/* eslint-disable no-undef */
import React from 'react';
import { createMount } from '@material-ui/core/test-utils';
import Button from '@material-ui/core/Button';
import { cleanup } from '@testing-library/react';
import ActionsMenu from '../ActionsMenu';

const actions = [
  {
    id: 1,
    name: 'Edit',
  },
  {
    id: 2,
    name: 'Book Accommodation',
  },
  {
    id: 3,
    name: 'Rate Accommodation',
  },
  {
    id: 4,
    name: 'Like/Dislike Accommodation',
  },
];

describe('<ActionsMenu />', () => {
  afterEach(() => cleanup);

  it('Should match the ActionsMenu snapshot', () => {
    const component = createMount()(
      <ActionsMenu actions={actions} />,
    );
    expect(component.html).toMatchSnapshot();
  });
});

describe('<ActionsMenu /> onClick()', () => {
  afterEach(() => cleanup);

  it('Calls the onClick handler to open', () => {
    const mockedHandleClick = jest.fn();
    const component = createMount()(
      <ActionsMenu actions={actions} />,
    );
    component.find(Button).first().props().onClick = mockedHandleClick;
    component.find(Button).first().props().onClick();

    expect(mockedHandleClick).toHaveBeenCalledTimes(1);
  });
});
