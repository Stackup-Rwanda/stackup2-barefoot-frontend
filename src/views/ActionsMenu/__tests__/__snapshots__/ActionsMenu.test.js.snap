// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionsMenu /> Should match the ActionsMenu snapshot 1`] = `[Function]`;
